module/documents/activity/summon.mjs

```diff
@@ -32,4 +42,612 @@ export default class SummonActivity extends ActivityMixin(SummonActivityData) {
+      if ( config ) {
+        actorUpdates["system.traits.size"] = size;
+        tokenUpdates.width = config.token ?? 1;
+        tokenUpdates.height = config.token ?? 1;
+      }
+    }
+
+    // Change creature type
+    if ( this.creatureTypes.size ) {
+      const type = this.creatureTypes.has(options.creatureType) ? options.creatureType : this.creatureTypes.first();
+      if ( actor.system.details?.race instanceof Item ) {
+        actorUpdates.items.push({ _id: actor.system.details.race.id, "system.type.value": type });
+      } else {
+        actorUpdates["system.details.type.value"] = type;
+      }
+    }
+
+    const attackDamageBonus = Roll.replaceFormulaData(this.bonuses.attackDamage, rollData);
+    const saveDamageBonus = Roll.replaceFormulaData(this.bonuses.saveDamage, rollData);
+    const healingBonus = Roll.replaceFormulaData(this.bonuses.healing, rollData);
+    for ( const item of actor.items ) {
+      const changes = [];
+
+      // Match attacks
+      if ( this.match.attacks && item.hasAttack ) {
+        const ability = this.item.abilityMod ?? rollData.attributes?.spellcasting;
+        const typeMapping = { mwak: "msak", rwak: "rsak" };
+        const parts = [
+          rollData.abilities?.[ability]?.mod,
+          prof,
+          rollData.bonuses?.[typeMapping[item.system.actionType] ?? item.system.actionType]?.attack
+        ].filter(p => p);
+        changes.push({
+          key: "system.attack.bonus",
+          mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE,
+          value: parts.join(" + ")
+        }, {
+          key: "system.attack.flat",
+          mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE,
+          value: true
+        });
+      }
+
+      // Match saves
+      if ( this.match.saves && item.hasSave ) changes.push({
+        key: "system.save.dc",
+        mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE,
+        value: rollData.item.save.dc ?? rollData.attributes.spelldc
+      }, {
+        key: "system.save.scaling",
+        mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE,
+        value: "flat"
+      });
+
+      // Damage bonus
+      let damageBonus;
+      if ( item.hasAttack ) damageBonus = attackDamageBonus;
+      else if ( item.system.actionType === "save" ) damageBonus = saveDamageBonus;
+      else if ( item.isHealing ) damageBonus = healingBonus;
+      if ( damageBonus && item.hasDamage ) changes.push({
+        key: "system.damage.parts",
+        mode: CONST.ACTIVE_EFFECT_MODES.ADD,
+        value: JSON.stringify([[`${damageBonus}`, ""]])
+      });
+
+      if ( changes.length ) {
+        const effect = (new ActiveEffect({
+          _id: staticID("dnd5eItemChanges"),
+          changes,
+          disabled: false,
+          icon: "icons/skills/melee/strike-slashes-orange.webp",
+          name: game.i18n.localize("DND5E.Summoning.ItemChanges.Label"),
+          origin: this.uuid,
+          flags: {
+            dnd5e: { type: "enchantment" }
+          }
```

Comment:
I guess we need a different solution for damage bonus now

```suggestion
          type: "enchantment"
```

Reply:
The whole item changes section will need to be reworked to handle activities.